pkg/verifier/aws/entry_point.go

```diff
@@ -243,3 +239,14 @@ func (a *AwsVerifier) VerifyDns(vdi verifier.VerifyDnsInput) *output.Output {
 
 	return &a.Output
 }
+
+// Cleans up the security groups created by network-verifier
+func CleanupSecurityGroup(vei verifier.ValidateEgressInput, a *AwsVerifier) *output.Output {
+	_, err := a.AwsClient.DeleteSecurityGroup(vei.Ctx, &ec2.DeleteSecurityGroupInput{GroupId: awsTools.String(vei.AWS.SecurityGroupId)})
+	if err != nil {
```

Comment:
Not exactly the same case, but it would be good to compare this with https://github.com/openshift/aws-vpce-operator/blob/main/controllers/vpcendpoint/cleanup.go#L133-L152 to not return an error if the error code from AWS is `InvalidGroup.NotFound`

Reply:
I agree we should add checks for that but I think it's out of the scope of this card.  The specific issue here is there are multiple places we return with an error where the security groups would be left dangling.  The current implementation will guarantee that CleanupSecurityGroup will always be called regardless of erroring out earlier.